includes/Plugin_Context.php

```diff
@@ -183,4 +191,35 @@ public function minimum_supported_wp() {
 
 		return $this->minimum_supported_wp;
 	}
+
+	/**
+	 * Gets the list of editable file names of the plugin.
+	 *
+	 * @since 1.1.0
+	 *
+	 * @return string[] Array of editable file names relative to the plugin root.
+	 */
+	public function editable_files() {
```

Comment:
Hmm not sure we really need a list of all editable files this way.

What do you think about `public function is_file_editable( string $file )` instead?

Reply:
No, what I meant is **only** `is_file_editable()`. No `editable_files` method.

Just something like this:

```php
public function is_file_editable( string $filename ) {
	$editable_extensions = wp_get_plugin_file_editable_extensions( $this->basename() );

	$extension  = strtolower( pathinfo( $filename, PATHINFO_EXTENSION ) );

	return in_array( $extension, $editable_extensions, true );
}
```

Reply:
Hmm, let me try this in my local setup.

Reply:
I tested in local setup. This will fix half of the issue. This does not handle the case of `no_plugin_readme`. Here in this case, there is no actual `readme.txt` file in the plugin. But the value of `filename` is still `readme.txt`. So edit link is still displayed for this error.

Reply:
So we can add a `file_exists` check?

Reply:
PR updated.